src/common/security/local/context.go

```diff
@@ -191,8 +203,35 @@ func (s *SecurityContext) GetProjectRoles(projectIDOrName interface{}) []int {
 			roles = append(roles, common.RoleGuest)
 		}
 	}
+	if len(roles) != 0 {
+		return roles
+	}
+	return s.GetRolesByGroup(projectIDOrName)
+}
 
-	//If len(roles)==0, Get Group Roles
-
+// GetRolesByGroup - Get the group role of current user to the project
+func (s *SecurityContext) GetRolesByGroup(projectIDOrName interface{}) []int {
+	var roles []int
+	user := s.user
+	project, err := s.pm.Get(projectIDOrName)
+	//No user, group or project info
+	if err != nil || project == nil || user == nil || len(user.GroupList) == 0 {
+		return roles
+	}
+	//Get role by LDAP group
+	groupDNConditions := group.GetGroupDNQueryCondition(user.GroupList)
+	roles = dao.GetRolesByLDAPGroup(project.ProjectID, groupDNConditions)
+	if err != nil {
+		return nil
+	}
 	return roles
 }
+
+// GetMyProjects ...
+func (s *SecurityContext) GetMyProjects() ([]*models.Project, error) {
```

Comment:
Seems the function doesn't return the projects which the user is a group member of.

Reply:
Updated